### DISCUSSION TITLE
wlan0, wlan1, eth0 & RaspAP

### DISCUSSION THREAD

**Discussion by mammique** [RaspAP/raspap-webgui#1315] [Q&A]
Hello, I'm in a not-so-usual case for which the documentation found online about bridging networks confuses me more than it helps. I've tried several configurations, most of the time my Rapberry Pi (4) no longer boots, I have to roll back to previous system backup and start over, without success so far. I have a WAN access coming on wlan0, I'm sharing it on my LAN over wlan1 via RaspAp. What I'm trying to do is to also have my Ethernet computers on the LAN to have access served by RaspAP on eth0 as it is already the case via wlan1 (DHCP, routing, etc.), and all computers on the eth0+wlan1 LAN being able to communicate with each other. Can anyone tell me how to achieve this?

Thanks you,

Bests,

Camille.

**Comment by billz** [comment]
Hi Camille,

RaspAP supports bridged configurations between the selected AP interface and wired ethernet (eth0). Behind the scenes, it writes to `/etc/dhcpcd.conf` like so:

```
# RaspAP br0 configuration
denyinterfaces eth0 wlan1
interface br0
```
Here wlan1 is taken from your example, but it can be any interface that supports AP mode.
The bridge interface (br0) is then specified along with the AP interface in `/etc/hostapd/hostapd.conf`:

```
interface=wlan1
bridge=br0
```

In your example, routing would need to occur between wlan0 and br0. This isn't supported via RaspAP's UI at the moment, however you could try manually updating hostapd.conf.

**Comment by mammique** [comment]
Hello, thank you very much for your reply, sorry for the feedback latency I was away from the Raspberry. I tried out you configuration, however it doesn't work, I had to roll back to previous config to be able to access wifi, as both wifi and ethernet were no longer working the new config. With the new config I had no DHCP reply from the Raspberry, neither over ethernet nor wifi. I knew I had no chance to get WAN access because you told me to manually modify hostapd.conf for routing (any tutorial BTW?), but I was expecting for a first step to get a DHCP answer on wifi or ethernet so I could communicate to the Raspberry from the client on the LAN, however nothing happened.

Here is the output of `ifconfig` (192.168.1.130 on wlan0 is WAN access provided by the Internet box) do you have any clue? Thank you very much.

```
br0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 169.254.124.237  netmask 255.255.0.0  broadcast 169.254.255.255
        inet6 fe80::a428:143e:8087:3c01  prefixlen 64  scopeid 0x20<link>
        ether c6:8f:9f:50:64:22  txqueuelen 1000  (Ethernet)
        RX packets 536  bytes 327379 (319.7 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 97  bytes 29519 (28.8 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet6 fe80::3700:d5da:ce49:5762  prefixlen 64  scopeid 0x20<link>
        ether e4:5f:01:bf:3f:a6  txqueuelen 1000  (Ethernet)
        RX packets 30  bytes 8472 (8.2 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 120  bytes 16402 (16.0 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Boucle locale)
        RX packets 5995  bytes 600676 (586.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 5995  bytes 600676 (586.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.130  netmask 255.255.255.0  broadcast 192.168.1.255
        inet6 2a02:8428:f01:eb01:69a7:397e:6d20:dcd1  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::8e81:d11a:c7eb:5e4c  prefixlen 64  scopeid 0x20<link>
        ether f4:a4:54:82:38:22  txqueuelen 1000  (Ethernet)
        RX packets 1789  bytes 1327637 (1.2 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 932  bytes 135045 (131.8 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.3.141.1  netmask 255.255.255.0  broadcast 10.3.141.255
        inet6 fe80::9f77:bba9:c5fe:1832  prefixlen 64  scopeid 0x20<link>
        ether bc:ec:43:44:1c:26  txqueuelen 1000  (Ethernet)
        RX packets 947  bytes 396771 (387.4 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 489  bytes 194774 (190.2 KiB)
        TX errors 0  dropped 102 overruns 0  carrier 0  collisions 0
```

**Reply by billz** [reply]
My response was meant to point you in the right direction - it wasn't a complete tutorial. 
Your question has been answered many times on [Raspberry Pi Stack Exchange](https://raspberrypi.stackexchange.com/).
Suggest you give a look there.

**Comment by mammique** [comment]
Hello, thank you for the reply, I browsed several topics on the Stack Exchange for an hour, the only posts I find on the topic are not using RaspAP, or outdated (old config files), or proposing to replace RaspAP by systemd-networkd. I'd like to keep RaspAp if possible, can you point me to a documentation about this case please? Thank you.

Bests,

Camille.

**Reply by billz** [reply]
I think you might be searching for the wrong topic. Your goals are beyond the current scope of RaspAP, so this is more a general Linux networking issue. Best advice is to follow the routed and bridged AP configurations in the [official RPi docs](https://www.raspberrypi.com/documentation/computers/configuration.html#setting-up-a-routed-wireless-access-point).

**Comment by mammique** [comment]
OK, thank you.

**Reply by mammique** [reply]
Update: This has been a nightmare, I also tried with systemd-networkd from scratch on a new distro following step by step tutorials, failed. Apparently there are also some kernel versions with bug on br0, and some hardware/firmware/driver flaws on some devices with AP mode. So finally I kept my old RaspAp install, and beside the wlan 10.3.141.1 subnet I just added an eth0 10.3.142.1 subnet (IP set in dhcpcd.conf and DHCP server settings in dnsmasq.conf), because of the already configured RapsAP routing all computers on both subnet can access eachother as well as WAN. Took 2 minutes, works at first try :-)
